app/grandchallenge/profiles/forms.py

```diff
@@ -74,6 +76,14 @@ def clean(self):
             # accounts with names fooAB barAB etc.
             raise ValidationError("Account details invalid")
 
+        username = self.cleaned_data.get("username", "")
+
+        if re.match(
+            r"(^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$)", username
+        ):
+            # standard python email regex, avoids the RFC nightmare email regex
```

Comment:
There is an existing validator that can be re-used, see https://docs.djangoproject.com/en/4.1/_modules/django/core/validators/#EmailValidator, though here we want the inverse.

Reply:
frustratingly, unlike the regex and url validator, it doesn't have an inverse - could use the regex validator with the above regex though - would that that be better?

Reply:
What about something like:

```python
try:
    EmailValidator()(username)
except ValidationError:
    # Username is not an email address
    pass
else:
    raise ValidationError("You cannot use an email address as a username")
```

Reply:
I'll take it - it is cleaner for sure

Reply:
I was always told that except...pass statements were somewhat unpythonic, but it's probably considerably more pleasant than a mystery regex

Reply:
That's fine in most cases, but here we're relying on the exception to do the thing we want to totally fine to suppress it.

Reply:
fixed this to use the EmailValidator - thanks for the pointers!